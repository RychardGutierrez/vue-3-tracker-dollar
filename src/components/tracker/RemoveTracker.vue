<template>
  <div @click="removeTracker"
    class="flex flex-col items-center gap-2 text-orange-600 cursor-pointer duration-500 hover:text-red-700">
    <img :src="iconDelete" alt="Remove Tracker" class="w-7" />
    <p>Remove</p>
  </div>
</template>

<script setup>
import iconDelete from '@/assets/delete.svg';

defineProps({
  removeTracker: {
    typeof: Function,
    require: true
  }
})

</script>

<style lang="scss" scoped>

</style>