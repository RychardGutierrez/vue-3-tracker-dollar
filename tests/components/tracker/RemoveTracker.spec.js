import { expect } from '@jest/globals';
import { flushPromises, shallowMount } from '@vue/test-utils';

import RemoveTracker from '../../../src/components/tracker/RemoveTracker.vue';

jest.mock('../../../src/assets/delete.svg', () => 'iconDelete');

describe('RemoveTracker component', () => {
  let wrapper = null;

  const jestRemoveTracker = jest.fn();

  beforeEach(() => {
    const props = {
      removeTracker: jestRemoveTracker,
    };

    wrapper = shallowMount(RemoveTracker, { props });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should to match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should click on Remove and call removeTracker method', async () => {
    const iconRemove = wrapper.find('div');

    await iconRemove.trigger('click');

    expect(jestRemoveTracker).toHaveBeenCalledTimes(1);
  });
});
