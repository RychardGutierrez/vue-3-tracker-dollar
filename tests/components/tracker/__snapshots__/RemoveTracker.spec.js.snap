// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveTracker component should to match with snapshot 1`] = `
"<div class=\\"flex flex-col items-center gap-2 text-orange-600 cursor-pointer duration-500 hover:text-red-700\\" removetracker=\\"function () {
        return fn.apply(this, arguments);
      }\\"><img src=\\"iconDelete\\" alt=\\"Remove Tracker\\" class=\\"w-7\\">
  <p>Remove</p>
</div>"
`;
